### DISCUSSION TITLE
Closing workers

### DISCUSSION THREAD

**Discussion by nambi-rajan-k** [anchan828/nest-bull#1800] [Q&A] [answered]
Hi , Is there a way to close a worker gracefully ? In bullmq it is done by calling close function through worker object like worker.close() , Is there something similar that can be done using this package ?

**Comment by anchan828** [comment, marked as answer]
@nambirajan-goxai Hi, You can use `BullService` to terminate queue/worker/event. However, since no care is taken after manual termination, unexpected problems may occur.

```ts
import { BullService } from "@anchan828/nest-bullmq";

export class YourService {
  constructor(private readonly bullService: BullService) {}

  public async closeWorker() {
    await this.bullService.workers[`queueName`].close();
  }
}
```
